### PR TITLE
Draft: Added support for Python models

### PR DESCRIPTION
**Pull request by RaunoArike**
This pull request adds support for generating counterfactuals for machine learning models defined in Python using PyTorch. This greatly extends the scope of machine learning models the package is compatible with, as the model doesn't have to be written in Julia anymore. This is particularly important given the widespread use of Python in the machine learning community.

Our implementation uses `PythonCall.jl` for the conversion and is agnostic to model architecture: any model defined using PyTorch is supported. The PR also includes a report that describes our investigation into `ONNX.jl` for solving this problem, the reasoning behind our decision to use `PythonCall.jl` instead, and our general approach and design choices when implementing this conversion.

### Changes made

```
Implemented a basic interface for models defined using PyTorch.
Created infrastructure that enables the user to import any model that has been defined using PyTorch with its weights saved in a Pickle file.
Wrote methods that made the PyTorch models compatible with the counterfactual generators in our package.
Documented the whole process, including a report on our investigation into `ONNX.jl` and the design choices made.
```

### Testing

```
Tested simple PyTorch models on all synthetic datasets currently used in the package.
```

### CODE REVIEW

File: docs/src/tutorials/model_catalogue.qmd [hunk 1]
```diff
@@ -51,6 +53,43 @@ end
 plot(plts...)
 ```
 
+## Importing PyTorch models
+
+The package supports generating counterfactuals for any neural network that has been previously defined and trained using PyTorch, regardless of the specific architectural details of the model. To generate counterfactuals for a PyTorch model, save the model inside a `.pt` file and call the following function:
```
Comment (kmariuszk) [inline docs/src/tutorials/model_catalogue.qmd:58]: Not fully sure about this part since we're defining a loss function in the `function ∂ℓ`:

`obj_loss = torch.nn.BCEWithLogitsLoss()(output, target)`.

What do you think, should we change the documentation or tune our implementation?

File: test/models.jl [hunk 2]
```diff
@@ -3,8 +3,11 @@ using CounterfactualExplanations.Models
 using Flux
 using LinearAlgebra
 using MLUtils
+using PythonCall
 using Random
 
+Random.seed!(0)
```
Comment (kmariuszk) [inline test/models.jl:9]: Do we actually need this?

**Review by kmariuszk — COMMENTED** [review]
@RaunoArike please take a look at my two comments, I think once we resolve them we can finally assign reviewers.

**Comment by RaunoArike** [comment]
Closed the pull request, as the functionality implemented here is now a part of PR #239
